### PR TITLE
Document bit fields usage

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Add a subchapter on how to use enums and sets to implement bit fields.

### CODE REVIEW

File: doc/sets_fragment.txt [hunk 1]
```diff
@@ -49,6 +49,34 @@ operation             meaning
...
 ==================    ========================================================
 
-Sets are often used to define a type for the *flags* of a procedure. This is
-a much cleaner (and type safe) solution than just defining integer
-constants that should be ``or``'ed together.
+Bit fields
+~~~~~~~~~~
+
+Sets are often used to define a type for the *flags* of a procedure or to implement
+bit fields for interoperability with C.
```
Comment (Araq) [inline doc/sets_fragment.txt]: No, for C interop we nowadays suggest a ``distinct cint`` type, C's enums are just too messy. Also: This should go into the FFI section.

Comment (mratsim) [reply doc/sets_fragment.txt]: The bit layout for Nim sets is incompatible anyway with traditional bit shifting for flags.

Got bitten by this when implementing mmap flags here: https://github.com/numforge/laser/blob/dbfb31d5b25c84d4264377eb6f5b876d6032c766/laser/photon_jit/photon_osalloc.nim#L43-L54

Comment (Araq) [reply doc/sets_fragment.txt]: The bit layout *can* be compatible but you have to understand how it works. :P

Comment (FedericoCeratto) [reply doc/sets_fragment.txt]: Updated.

**Pull request merged**